test/test_pipeline/test_classification.py

```diff
@@ -140,54 +162,88 @@ def test_find_preprocessors(self):
-            auto.predict_proba(X_test)
+        """Test that seeded SimpleClassificaitonPipeline returns good results on iris
+
+        Expects
+        -------
+        * The performance of configuration with fixed seed gets above 96% accuracy on iris
+        """
+        X_train, Y_train, X_test, Y_test = get_dataset(dataset='iris')
+
+        auto = SimpleClassificationPipeline(random_state=1)
+
+        auto = auto.fit(X_train, Y_train)
+        predictions = auto.predict(X_test)
+
+        acc = sklearn.metrics.accuracy_score(predictions, Y_test)
+        self.assertAlmostEqual(0.96, acc)
 
     def test_default_configuration_multilabel(self):
-        for i in range(2):
-            classifier = SimpleClassificationPipeline(
-                random_state=1,
-                dataset_properties={'multilabel': True}
-            )
-            cs = classifier.get_hyperparameter_search_space()
-            default = cs.get_default_configuration()
-            X_train, Y_train, X_test, Y_test = get_dataset(dataset='iris',
-                                                           make_multilabel=True)
-            classifier.set_hyperparameters(default)
-            classifier = classifier.fit(X_train, Y_train)
-            predictions = classifier.predict(X_test)
-            self.assertAlmostEqual(0.96,
-                                   sklearn.metrics.accuracy_score(predictions,
-                                                                  Y_test))
-            classifier.predict_proba(X_test)
+        """Test that SimpleClassificationPipeline default config returns good results on
+        a multilabel version of iris.
+
+        Expects
+        -------
+        * The performance of a random configuratino gets above 96% on a multilabel
+            version of iris
+        """
+        X_train, Y_train, X_test, Y_test = get_dataset(dataset='iris', make_multilabel=True)
+
+        classifier = SimpleClassificationPipeline(dataset_properties={'multilabel': True})
+        cs = classifier.get_hyperparameter_search_space()
+
+        default = cs.get_default_configuration()
+        classifier.set_hyperparameters(default)
+
+        classifier = classifier.fit(X_train, Y_train)
+        predictions = classifier.predict(X_test)
+
+        acc = sklearn.metrics.accuracy_score(predictions, Y_test)
+        self.assertAlmostEqual(0.96, acc)
 
     def test_default_configuration_iterative_fit(self):
+        """Test that the SimpleClassificationPipeline default config for random forest
+        with no preprocessing can be iteratively fit on iris.
+
+        Expects
+        -------
+        * Random forest pipeline can be fit iteratively
+        * Test that its number of estimators is equal to the iteration count
+        """
+        X_train, Y_train, X_test, Y_test = get_dataset(dataset='iris')
+
         classifier = SimpleClassificationPipeline(
-            random_state=1,
```

Comment:
This is now inconsistent. The `random_state` is not dropped in a few unit tests above.

Reply:
That's because this test relies an an accuracy score, it needs to be fixed as not all configurations will get 96%. The other tests are not specifically for metrics.